autoload/db.vim

```diff
@@ -519,3 +602,129 @@ function! db#clobber_dbext(...) abort
   endfor
   return opts
 endfunction
+
+function! s:vim_job.cb(job, data) dict abort
+  if type(a:data) ==? type(0)
+    let self.exit = 1
+    let self.exit_status = a:data
+    return self.call_cb_if_finished()
+  endif
+  let self.output .= a:data
+endfunction
+
+function! s:vim_job.close_cb(channel) dict abort
+  let self.close = 1
+  return self.call_cb_if_finished()
+endfunction
+
+function! s:vim_job.call_cb_if_finished() abort
+  if self.close && self.exit
+    return self.callback(split(self.output, "\n", 1), self.exit_status)
+  endif
+endfunction
+
+function! s:nvim_job_cb(jobid, data, event) dict abort
+  if a:event ==? 'exit'
+    return self.callback(self.output, a:data)
+  endif
+  call extend(self.output, a:data)
+endfunction
+
+function! s:job_run(cmd, callback, stdin_file, ...) abort
+  let stdin = get(a:, 1, '')
+  if has('nvim')
+    let jobid = jobstart(a:cmd, {
+          \ 'on_stdout': function('s:nvim_job_cb'),
+          \ 'on_stderr': function('s:nvim_job_cb'),
+          \ 'on_exit': function('s:nvim_job_cb'),
+          \ 'output': [],
```

Comment:
This should be `['']`, and the final `['']` recieved on both stdout and stderr should be ignored. Otherwise, there's an extra blank line in the final list, which was the root cause of #116.

Reply:
This commit [tpope/vim-dadbod@`7b05a3c` (#117)](https://github.com/tpope/vim-dadbod/pull/117/commits/7b05a3c5dd2d0842f6e0c47aa4f38f7d41f48a56) should solve this issue. I can just move that logic from there to here.

Reply:
It makes sense to move it here since I would not need to duplicate the logic on 2 places. I just don't understand why we need `['']` as initial value?

Reply:
Oh, you need to be doing this:

https://github.com/neovim/neovim/blob/d367ed9b23d481998d297d812f54b950e5511c24/runtime/doc/channel.txt#L93-L100

Otherwise, you'll get an extra line break in between every callback. You probably haven't seen that much in practice because db clients tend to dump everything in one go.

And that also completely solves #116. You don't need to ignore the final `['']` like I proposed because concatenating that empty string is a no-op.

Reply:
Actually I forgot about `stdout_buffered`, that renders most of my remark obsolete. It still would fix the problem, but it's not the only way to do so.